soccer/gameplay/plays/testing/four_courners_pass.py

```diff
@@ -0,0 +1,229 @@
+import robocup
+import play
+import behavior
+import skills.move
+import skills.capture
+import tactics.coordinated_pass
+import constants
+import main
+import enum
+
+
+## A testing play to demonstrate our ability to pass and recieve balls
+# One robot will pursue the ball while three other robots will pass the ball amongst themselves
+# they can only pass the ball at the corners and recieve at the corners and they cannot move
+# through the center of the square
+class FourCornerPass(play.Play):
+	class State(enum.Enum):
+		## One robot goes captures the ball while the other two gets on corners
+		# pursuing robot does not move on the first instance
+		setup = 1
+
+		## The robots pass to each other and begin setting up the next pass
+		# while the chasing robot goes for the ball.
+		passing = 2
```

Comment:
In terms of states, the current setup uses a move and capture to get the robots into the first two points and get the ball. Then the robot picks someone to pass to, passes there, and commands the second offensive robot to move to the next corner.

Pass has a built in capture and move for the target robots. I think we could do this where the state is the pass and movement to the next corner. The start might be a little jank looking, but we could have a true setup state where we pass the ball into the first corner instead of starting where ever the ball started. This way there is no chance that the robots change between the capture and pass. This play can transition to itself whenever the pass completes. If you want finer control of the defender during this point, you can have a second state between the passes to mess with him.

This will also allow only a single move subbehavior for the offense and a pass. This will change the behavior so we may not have the ball under control for the pass before we start moving the second robot. If that's the case, having an intermediate state where we transition after the start the kick part of the coordinated pass may be a good idea.

Reply:
I think a cool idea would be a modification of coordinated pass that implements the 'premove' idea that Tanner implemented here, where the passer aiming and the receiver getting into position happen concurrently